### ISSUE TITLE
Rule S2053: issue message is not correct when salt is too short

### ISSUE THREAD

**Issue by pierre-loup-tristant-sonarsource** [SonarSource/sonar-dotnet#4289]
In [smsportal-msbuild14](https://peach.sonarsource.com/issues?projects=smsportal-msbuild14&resolved=false&rules=csharpsquid%3AS2053) we raise 2 TPs but the fix suggested by the issue message is incorrect:

>Make this salt longer.

Making the salt longer will not fix the issue as the main problem is that the salt is hardcoded ans therefore predictable.
We should only advice to use a longer salt in the case where the size is byte array size is shorter that 16 *and* it's populated from a cryptographically secure random function.

Also the message has been updated the to "* Make this salt at least 16 bytes.".

```csharp
var shortSalt = new byte[8];
RandomNumberGenerator.Create().GetBytes(shortSalt);
var pdbShortSalt = new PasswordDeriveBytes(passwordString, shortSalt);          // Noncompliant: Make this salt at least 16 bytes

var pdbPredictableSalt = new PasswordDeriveBytes(passwordString, new byte[8]);          // Noncompliant: Make this salt unpredictable.
```
